cosmwasm/get-p2p-tokens.ts

```diff
@@ -0,0 +1,299 @@
+    }
+    throw new Error(`Failed to execute function after ${MAX_RETRIES} retries: ${lastError}`);
+}
+
+async function getTokensFromBlock(
+    axelarChainId: string,
+    its: Contract,
+    filter: providers.Filter,
+    startBlockNumber: number,
+    eventsLength: number,
+    max: number,
+    provider: providers.Provider,
+): Promise<SquidTokenDataWithTokenId[]> {
+    const end = Math.min(startBlockNumber + eventsLength, max);
+
+    if (startBlockNumber > end) {
+        return [];
+    }
+
+    const events = await runWithRetries(async () => await its.queryFilter(filter, startBlockNumber, end));
+    const tokens = await runWithRetries(async () => {
+        const tokenData: SquidTokenDataWithTokenId[] = await Promise.all(
+            events
+                .map((event) => event.args)
+                .map(async (event): Promise<SquidTokenDataWithTokenId> => {
+                    const tokenId = event[0];
+                    const tokenManagerAddress = event[1];
+                    const tokenManagerType = event[2];
+                    const tokenInfo = await getTokenInfo(tokenManagerAddress, tokenManagerType, provider);
+                    const interchainTokenAddress = await its.interchainTokenAddress(tokenId);
+
+                    if (interchainTokenAddress !== tokenInfo.tokenAddress && tokenManagerType === 0) {
+                        printWarn(
+                            `Token ${tokenId} is conflicting for ${axelarChainId} with interchain token address ${interchainTokenAddress}`,
+                        );
+                    }
+
+                    return {
+                        axelarChainId,
+                        tokenId,
+                        tokenManager: tokenManagerAddress,
+                        tokenManagerType: getTokenManagerTypeString(tokenManagerType) as SquidTokenManagerType,
+                        conflictingInterchainTokenAddress:
+                            interchainTokenAddress !== tokenInfo.tokenAddress && tokenManagerType === 0 ? interchainTokenAddress : null,
+                        ...tokenInfo,
+                    } as SquidTokenDataWithTokenId;
+                }),
+        );
+        return tokenData;
+    });
+    return tokens;
+}
+
+async function getTokensFromChain(name, chainInfo, tokensInfo: SquidTokenInfoFileWithChains) {
+    try {
+        if (!tokensInfo.chains[name]) {
+            tokensInfo.chains[name] = {
+                start: 0,
+                end: 0,
+                max: 0,
+                alreadyProcessedPercentage: '0.00',
+                rpcs: [chainInfo.rpc],
+            };
+        }
+        const currentChain = tokensInfo.chains[name];
+        if (!chainInfo.contracts.InterchainTokenService || chainInfo.contracts.InterchainTokenService.skip) {
+            return;
+        }
+        printInfo(`ITS at ${name} is at`, chainInfo.contracts.InterchainTokenService.address);
+
+        const eventsLength = queryLimit[name.toLowerCase()] || 2048;
```

Comment:
to get this PR merged faster, and to keep focus on the squid config (vs adding too many things), this script can be added in another PR. This will be needed for registering all deployed tokens on a chain, but don't want to get bogged down in reviewing this code that's trickier to handle